### PR TITLE
fix: Fix tests when running with Rust normalizer

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: src/sentry/models/event.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def get_interfaces(self):
     def interfaces(self):
         return self.get_interfaces()
 
-    def get_tags(self):
+    def get_tags(self, sorted=True):
```
Comment (untitaker) [inline src/sentry/models/event.py]: undo this

File: src/sentry/api/serializers/models/event.py [hunk 2]
```diff
@@ -96,7 +96,8 @@ def _get_tags_with_meta(self, event):
                     '_meta': meta.get(kv[0]) or get_path(meta, six.text_type(i), '1') or None,
                 }
                 for i, kv in enumerate(event.data.get('tags') or ())
-                if kv is not None and kv[0] is not None and kv[1] is not None],
+                if kv is not None and kv[0] is not None and kv[1] is not None
```
Comment (untitaker) [inline src/sentry/api/serializers/models/event.py]: undo this

File: src/sentry/testutils/fixtures.py [hunk 3]
```diff
@@ -522,6 +522,19 @@ def create_event(self, event_id=None, normalize=True, **kwargs):
         event.save()
         return event
 
+    def create_event_endtoend(self, data, project_id, assert_no_errors=True):
```
Comment (untitaker) [inline src/sentry/testutils/fixtures.py]: Deprecate `create_event`

File: src/sentry/testutils/fixtures.py [hunk 4]
```diff
@@ -508,9 +509,6 @@ def create_event(self, event_id=None, normalize=True, **kwargs):
             kwargs['data'] = manager.get_data()
             kwargs['message'] = manager.get_search_message()
 
-        else:
-            assert 'message' not in kwargs, 'do not pass message this way'
```
Comment (untitaker) [inline src/sentry/testutils/fixtures.py]: @mitsuhiko this didnt actually work, message was always passed this way because there was a setdefault above

Comment (mitsuhiko) [reply src/sentry/testutils/fixtures.py]: Might be worth checking on the search_message PR what happens there.

File: src/sentry/testutils/fixtures.py [hunk 5]
```diff
@@ -466,6 +466,7 @@ def create_useremail(self, user, email, **kwargs):
         return useremail
 
     def create_event(self, event_id=None, normalize=True, **kwargs):
+        # XXX: Do not use this method for new tests! Prefer `store_event`.
```
Comment (jan-auer) [inline src/sentry/testutils/fixtures.py]: Can we emit a warning to stdout/stderr if this is being called directly from a test?

File: tests/sentry/api/endpoints/test_project_user_reports.py [hunk 6]
```diff
@@ -209,18 +209,17 @@ def test_with_dsn_auth_invalid_project(self):
...
         self.login_as(user=self.user)
-
         project = self.create_project()
-        group = self.create_group(project=project)
         environment = self.make_environment(project)
         event = self.create_event(
-            group=group,
-            tags={'environment': environment.name},
-            datetime=timezone.now(),
+            data={
```
Comment (jan-auer) [inline tests/sentry/api/endpoints/test_project_user_reports.py]: Is this intentionally still using `create_event`?

File: tests/sentry/web/api/tests.py [hunk 7]
```diff
@@ -464,12 +465,20 @@ def test_scrub_data_off(self, mock_insert_data_to_database):
...
-            'foo': ['1'],
-            'bar': ['2'],
-            'baz': ['3']
-        }
+        if not ENABLE_RUST:
+            assert call_data['request']['data'] == {
+                'password': ['lol'],
+                'foo': ['1'],
+                'bar': ['2'],
+                'baz': ['3']
```
Comment (jan-auer) [inline tests/sentry/web/api/tests.py]: Thank god ❤️

Comment (jan-auer) [reply tests/sentry/web/api/tests.py]: I mean, thank you!

File: src/sentry/testutils/fixtures.py (hunk 5 above)
Comment (untitaker) [reply src/sentry/testutils/fixtures.py]: Most tests still use this. I am not sure how to fix this situation. I would propose requiring a kwarg `is_legacy_test=True` to deter from using this in new tests.

Migrating the entire testsuite is unlikely to ever happen IMO.

File: tests/sentry/api/endpoints/test_project_user_reports.py (hunk 6 above)
Comment (untitaker) [reply tests/sentry/api/endpoints/test_project_user_reports.py]: No, I was just very lazy with porting. Will fix though, this seems like a half-way port

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**